src/include/kdb.h.in

```diff
@@ -134,80 +125,61 @@ int kdbSet (KDB *handle, KeySet *returned,
  **************************************/
 
 /* Basic Methods */
-Key *keyNew (const char *keyname, ...) ELEKTRA_SENTINEL;
-Key *keyVNew (const char *keyname, va_list ap);
-
-Key * keyCopy (Key *dest, const Key *source, elektraCopyFlags flags);
-
-int keyClear (Key *key);
-int keyDel (Key *key);
-
-uint16_t keyIncRef (Key *key);
-uint16_t keyDecRef (Key *key);
-uint16_t keyGetRef (const Key *key);
+ElektraKey * elektraKeyNew (const char * keyname, ...) ELEKTRA_SENTINEL;
+ElektraKey * elektraKeyVNew (const char * keyname, va_list ap);
 
-/* Meta Info */
-int keyRewindMeta (Key *key);
-const Key *keyNextMeta (Key *key);
-const Key *keyCurrentMeta (const Key *key);
+ElektraKey * elektraKeyCopy (ElektraKey * dest, const ElektraKey * source, elektraCopyFlags flags);
 
-int keyCopyMeta (Key *dest, const Key *source, const char *metaName);
-int keyCopyAllMeta (Key *dest, const Key *source);
+int elektraKeyClear (ElektraKey * key);
+int elektraKeyDel (ElektraKey * key);
 
-const Key *keyGetMeta (const Key *key, const char* metaName);
-ssize_t    keySetMeta (Key *key, const char* metaName,
-	const char *newMetaString);
-KeySet * keyMeta (Key * key);
+uint16_t elektraKeyIncRef (ElektraKey * key);
+uint16_t elektraKeyDecRef (ElektraKey * key);
+uint16_t elektraKeyGetRef (const ElektraKey * key);
 
-/* Methods for Making Tests */
-int keyCmp (const Key *k1, const Key *k2);
+/* Metadata */
+ElektraKeySet * elektraKeyMeta (ElektraKey * key);
+int elektraKeySetMeta (ElektraKey * key, ElektraKeySet * meta);
 
-int keyNeedSync (const Key *key);
+/* Relations */
+int elektraKeyCompareName (const ElektraKey * k1, const ElektraKey * k2);
 
-int keyIsBelow (const Key *key, const Key *check);
-int keyIsBelowOrSame (const Key *key, const Key *check);
-int keyIsDirectlyBelow (const Key *key, const Key *check);
-
-int keyIsBinary (const Key *key);
-int keyIsString (const Key *key);
+int elektraKeyIsBelow (const ElektraKey * key, const ElektraKey * check);
+int elektraKeyIsBelowOrSame (const ElektraKey * key, const ElektraKey * check);
+int elektraKeyIsDirectlyBelow (const ElektraKey * key, const ElektraKey * check);
```

Comment:
```suggestion
```

should be moved to operations lib and merged (see #4245)

Reply:
I agree about the merging, not sure if it can be moved to operations (cyclic links to ksCut)

Reply:
IMO this is more a question of "Is `ksCut` minimal enough for the core?"

Reply:
If ksCut is the only thing that depends then case is quite clear: then we move elektraKeyIsBelow and elektraKsCut (or elektraKeysetCut?) to the operations lib.